rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -823,17 +827,25 @@ def predict_label(
 
         # if X contains all zeros do not predict some label
         if label_ids.size > 0:
-            label = {
-                "name": self.inverted_label_dict[label_ids[0]],
-                "confidence": message_sim[0],
-            }
 
+            # normalise scores if turned on
+            if self.loss_type == "softmax" and self.ranking_length > 0:
+                ranking_length = self.ranking_length
+                label_ids = label_ids[:ranking_length]
+                message_sim = message_sim[:ranking_length] / (
+                    np.sum(message_sim[:ranking_length])
+                )
+
+            message_sim = message_sim[:LABEL_RANKING_LENGTH]
             ranking = list(zip(list(label_ids), message_sim))
-            ranking = ranking[:LABEL_RANKING_LENGTH]
```

Comment:
keep this one

Reply:
then the `ranking_length: 3` will not truncate properly

Reply:
the way it currently works is that it truncates based on ranking length if it was normalized, then in also truncates based on label ranking length. therefore the lowest will always win

Reply:
```suggestion
            if ranking_length > LABEL_RANKING_LENGTH:
                ranking_length = LABEL_RANKING_LENGTH 
            ranking = ranking[: ranking_length]
```

Reply:
so you can normalize first. Then truncate all together, this way it easier to modify behavior in the future

Reply:
because anyhow you need to truncate if it is not softmax

Reply:
it seems to me the current version will raise an error if loss type is not softmax. Because you don't truncate `label_ids`. Also, it seems to me that if `ranking_length` is higher than `LABEL_RANKING_LENGTH`, then `label_ids` and `message_sim` are truncated to different lengths?

Reply:
> Because you don't truncate label_ids

Yeah I did miss that when I undid the last changes, I see. `zip` takes care of it and avoids the error I think. In any case I get what you mean, i'll make a numpy util and use it in `calculate_message_sim`